### PR TITLE
iOS interface with new scene Update API

### PR DESCRIPTION
**Pull request by karimnaaji**
Follows https://github.com/tangrams/tangram-es/pull/1125
Steps towards https://github.com/tangrams/tangram-es/issues/1063

### CODE REVIEW

File: ios/src/TGSceneUpdate.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//
+//  Created by Karim Naaji on 11/05/16.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+@interface TGSceneUpdate : NSObject
+
+- (id)initWithPath:(NSString *)path value:(NSString *)value;
```
Comment (msmollin) [inline ios/src/TGSceneUpdate.h]: `instancetype` instead of `id` and mark as nullable/nonnull

File: ios/src/TGSceneUpdate.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@property (strong, nonatomic) NSString* path;
```
Comment (msmollin) [inline ios/src/TGSceneUpdate.h]: these should be 
`+@property (copy, nonatomic) NSString* path;` due to how NSStrings are implemented under the covers.

File: ios/src/TGSceneUpdate.mm [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+//
+//  Created by Karim Naaji on 11/05/16.
+//
+//
+
+#import "TGSceneUpdate.h"
+
+@implementation TGSceneUpdate
+
+- (id)initWithPath:(NSString *)path value:(NSString *)value;
```
Comment (msmollin) [inline ios/src/TGSceneUpdate.mm]: instancetype here as well.

**Comment by karimnaaji** [comment]
@msmollin Updated

**Comment by karimnaaji** [comment]
@tallytalwar can you gives this a review?

**Comment by tallytalwar** [comment]
Will get to it today.

**Pull request merged**